addons/binding/org.openhab.binding.osramlightify/README.md

```diff
@@ -0,0 +1,125 @@
+
+After adding gateways, they are automatically scanned for paired devices and current device states immediately and then periodically according to the polling interval set in the configuration for the thing representing the gateway.
+
+
+## Notes
+
+### Power on/off
+
+When devices are powered on it typically takes several seconds for them to join the ZigBee network and for the gateway poll to see them and bring the things online in openHAB. If multiple devices are powered on at the same time they join the network one at a time serially so it can take quite a while until everything is done.
+
+When devices are powered off it takes 10 minutes for them to go offline in openHAB. This is a limitation of the gateway and protocol(s) involved.
+
+### Pairing
+
+When pairing devices with a gateway using the Lightify app sometimes the app appears to have a name for the device but the gateway does not actually have it set and so the device shows up in the openHAB inbox with a generic name. This is not a problem with openHAB or the OSRAM Lightify binding. If this happens you can correct the gateway via the Lightify app by tapping on the device, tapping on the name at the top of the screen, then tapping done to write the name to the gateway. It does not appear to be necessary to change the name at all. OpenHAB will then update the inbox entry with the new name on the next poll (you may need to reload paperUI to see it though).
+
+
+## Channels
+
+The OSRAM Lightify supports the following channels (although not all channels are available on all devices):
+
+| Channel            | Item Type    | Description                                           |
+|--------------------|--------------|-------------------------------------------------------|
+| color              | Color        | Changes the color                                     |
+| switch             | Switch       | Switch the device on/off                              |
+| dimmer             | Dimmer       | Set the brightness between 0% and 100%                |
+| temperature        | Dimmer       | Set the white temperature between 0% and 100%         |
+| absTemperature     | Number       | Set the white temperature to an absolute Kelvin value |
+
+Note that the temperature and absTemperature channels are linked so that changing one changes the other as well.
+
+
+## Full Example
+
+Textual configuration is not supported. Lightify devices do not have any markings that would allow them to be added manually. They **MUST** be paired with a gateway and the gateway **MUST** be queried to find out how to address them. It would in theory be possible to then take this information and to create a suitable `.things` file however this has not been tested.
```

Comment:
It is simply not possible. The addresses are kind of MAC addresses of the paired bulbs / devices. The gateway needs to be queried to get those addresses and be able to address them (all configuration goes through the gateway anyways). The only thing that would be possible to configure manually (and I doubt it is meaningful), would be the IP of the gateway but it's actually more convenient / meaningful to discover it automatically.

If every binding needs to have textual configuration, the design is broken :) my 2.5 cents

Reply:
> If every binding needs to have textual configuration, the design is broken :)

Nope, the other way round - if a binding is not following the architecture, the binding is broken. It does not have to be simple to get hold of the information (you could e.g. use https://github.com/eclipse/smarthome/issues/4817), but it must be possible to define it textually.

Reply:
We agree to disagree, that's a start. I say the architecture defines something which is not possible for all the possible situations. It might be possible to address an arbitrary device and send that to the gateway but it either will fail and restart or discard the message if not paired (I hope for the latter though). So what should be the meaning of textually defining a thing you can't use or address because it's probably not paired.

If it is just about serializing a textual representation down to disk, this is different from manually creating a thing. Anyhow I still can't see how this rule makes sense in a general term.

Reply:
PS: @triller-telekom don't get me wrong, I bet @mjagdis appreciates the comments (as do I).

Reply:
I agree with @kaikreuzer: A binding should always provide a textual configuration as well. If it is hard to create such a configuration is a different topic. But a user should be able to, for example, use your auto discovery to find out these IEEE addresses and afterwards create a textual configuration where he also can name its things and items as he likes, maybe he doesn't like your automatic name creation?